### PR TITLE
705 log slow oplog

### PR DESCRIPTION
**Pull request by savinay-vijay**
**Description of changes**

The —oplog flag has been added to the argument parser for mplotqueries. The parser, on receiving the --oplog flag, searches for loglines containing  the “applied op”  token (for reference, see https://docs.mongodb.com/manual/core/replica-set-oplog/). On encountering the token, the parser uses the built-in duration function to parse the duration and to plot it on a scatter-plot.
For more accuracy, the parser can also verify if the operation in the log line is an “REPL” operation (For reference: https://docs.mongodb.com/manual/core/replica-set-oplog/).

**Testing**
The following testing tools have been used to test the newly added code:

1. Tox
2. Flake8
3. iSort
4. Doc8

2. The test result of a code coverage can be seen in table below:

**Name                                              Stmts   Miss  Cover
---------------------------------------------------------------------**
mtools/mplotqueries/mplotqueries.py                 400     98    76%
mtools/util/logevent.py                             507      7    99%

**Version Information**
| macOS 		| 10.14.5	|
| mongod		| v4.0.10	|
| MongoDB shell	| v4.0.10	|

### CODE REVIEW

**Review by HenryGP — APPROVED** [review]
Super cool that you are testing the mplotqueries entries :)

File: mtools/test/test_mplotqueries.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        self.tool = MPlotQueriesTool()
+        self._test_init()
+
+    def _test_init(self, filename='mongod_225.log'):
+        # load logfile(s)
+        self.logfile_path = os.path.join(os.path.dirname(mtools.__file__),'test/logfiles/', filename)
+        self.logfile = LogFile(open(self.logfile_path, 'rb'))
+
+    def test_oplog(self):
+        #different logfile for oplogs
```
Comment (thiagogalesi) [inline mtools/test/test_mplotqueries.py]: missing a space between # and the rest of the comment ;)

File: mtools/test/test_mplotqueries.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        logfile_oplog = "mtools/test/logfiles/mongod.log"
+        self.tool.run('%s --oplog --group operation' % logfile_oplog)
+        output = sys.stdout.getvalue()
+        lines = output.splitlines()
+        assert any(map(lambda line: 'SCATTER plot' in line, lines))
```
Comment (thiagogalesi) [inline mtools/test/test_mplotqueries.py]: while this is fine, it's better to start using generator expressions (or list comprehension)

```suggestion
        assert any('SCATTER plot' in line for line in lines)
```

**Review by thiagogalesi — COMMENTED** [review]
two minor things

**Review by davidedeangelismdb — APPROVED** [review]

**Review by thiagogalesi — APPROVED** [review]

**Review by manfontan — APPROVED** [review]

**Pull request merged**